lib/read.js

```diff
@@ -179,3 +185,14 @@ function contentstream (req, debug, inflate) {
 
   return stream
 }
+
+function preserveAsyncContext (asyncResourceType, fn) {
+  if (!asyncHooks) {
+    return fn
+  }
+  var asyncResource = new asyncHooks.AsyncResource(asyncResourceType)
+  return function () {
+    var args = [].slice.call(arguments)
+    asyncResource.runInAsyncScope.apply(asyncResource, [fn, null].concat(args))
```

Comment:
Rather than this apply here, you could replace the whole function this is wrapped in by using bind. Something like return asyncResource.runInAsyncScope.bind(asyncResource, fn)

Reply:
Oh, that's way nicer :slightly_smiling_face:. Thanks.